src/test/lib/article/tag-utils.test.ts

```
@@ -0,0 +1,191 @@
+    expect(filteredPosts[0].data.title).toBe("Best Practices for React");
+  });
+
+  it("should filter posts by Optimization tag", () => {
+    const filteredPosts = filterPostsByTag(mockPosts as any, "Optimization", mockTagTranslations);
+    
+    expect(filteredPosts).toHaveLength(1);
+    expect(filteredPosts[0].data.title).toBe("API Optimization Tips");
+  });
+
+  it("should return empty array for non-existent tag", () => {
+    const filteredPosts = filterPostsByTag(mockPosts as any, "Non-existent", mockTagTranslations);
```

ESLint

[error] Unexpected any. Specify a different type.

(@typescript-eslint/no-explicit-any)